mm2src/coins/utxo/rpc_clients/electrum_rpc/client.rs

```diff
@@ -0,0 +1,990 @@
+
+    fn client_info(&self) -> String { UtxoJsonRpcClientInfo::client_info(self) }
+
+    fn transport(&self, request: JsonRpcRequestEnum) -> JsonRpcResponseFut {
+        Box::new(self.clone().electrum_request_multi(request).boxed().compat())
+    }
+}
+
+impl JsonRpcBatchClient for ElectrumClient {}
+
+impl JsonRpcMultiClient for ElectrumClient {
+    fn transport_exact(&self, to_addr: String, request: JsonRpcRequestEnum) -> JsonRpcResponseFut {
+        Box::new(
+            self.clone()
+                .electrum_request_to(to_addr.clone(), request)
+                .map_ok(|response| (JsonRpcRemoteAddr(to_addr), response))
+                .boxed()
+                .compat(),
+        )
+    }
+}
+
+#[cfg_attr(test, mockable)]
+impl ElectrumClient {
+    pub async fn try_new(
+        client_settings: ElectrumClientSettings,
+        event_handlers: Vec<Box<SharableRpcTransportEventHandler>>,
+        block_headers_storage: BlockHeaderStorage,
+        abortable_system: AbortableQueue,
+        scripthash_notification_sender: Option<UnboundedSender<ScripthashNotification>>,
+    ) -> Result<ElectrumClient, String> {
+        let client = ElectrumClient(ElectrumClientImpl::try_new_arc(
+            client_settings,
+            block_headers_storage,
+            abortable_system,
+            event_handlers,
+            scripthash_notification_sender,
+        )?);
+
+        // Wait till we have some connection(s) available.
+        client
+            .connection_manager
+            .wait_till_connected(5.)
+            .await
+            // Connections might be temporarily unavailable, so we don't want to fail the client creation.
+            .map_err(|e| error!("{:?}", e))
+            .ok();
+
+        Ok(client)
+    }
+
+    /// Sends a JSONRPC request to all the connected servers.
+    ///
+    /// A client with `ConnectionManagerPolicy::Multiple` will send the request to all active connected servers,
+    /// which are *all* the servers if non of them is erroring (timeout, version mismatch, etc).
+    /// A client with `ConnectionManagerPolicy::Selective` will send the request to the currently selected active server.
+    async fn electrum_request_multi(
+        self,
+        request: JsonRpcRequestEnum,
+    ) -> Result<(JsonRpcRemoteAddr, JsonRpcResponseEnum), JsonRpcErrorType> {
+        // Whether to send the request to all the connections or not.
+        let send_to_all = matches!(request, JsonRpcRequestEnum::Single(ref req) if req.method == "server.ping");
+        // Request id and serialized request.
+        let req_id = request.rpc_id();
+        let request = json::to_string(&request).map_err(|e| JsonRpcErrorType::InvalidRequest(e.to_string()))?;
+
+        let connections = self.connection_manager.get_active_connections().await;
+        if connections.is_empty() {
+            return Err(JsonRpcErrorType::Transport("No connections available".to_string()));
+        }
```

Comment:
It would, we can wait a little bit here to make things more robust. But that won't eliminate such an edge case completely.
We will want to retry the request in the swaps etc... too.

Reply:
> But that won't eliminate such an edge case completely.

Yes, we can't eliminate the edge case of when all electrums are down, but we need to make selective mode as efficient as legacy multiple mode otherwise users will go back to multiple mode as time goes by. For multiple mode, all electrums were tried sequentially, and this should be the case for selective as well but without keeping all connections open.